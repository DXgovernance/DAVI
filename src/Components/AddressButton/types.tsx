export interface AddressButtonProps {
  address?: string;
  transactionsCounter?: number;
  onClick?: () => void;
  showFullAddress?: boolean;
}
