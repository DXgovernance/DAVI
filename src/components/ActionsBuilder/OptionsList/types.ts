import { Option } from '../types';
export interface OptionsListProps {
  isEditable: boolean;
  options: Option[];
  onChange: (options: Option[]) => void;
  addOption: () => void;
  editOption: (option: Option) => void;
}

export enum SimulationState {
  none,
  pending,
  allPassed,
  someFailed,
  error,
}
