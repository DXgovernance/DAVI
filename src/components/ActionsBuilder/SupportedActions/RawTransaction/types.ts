import { BigNumber } from 'ethers';

export interface RawTransactionValues {
  to: string;
  value: BigNumber;
  data: string;
}
