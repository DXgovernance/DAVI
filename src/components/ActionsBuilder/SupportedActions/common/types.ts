import { DecodedCall } from 'components/ActionsBuilder/types';

export interface SummaryProps {
  decodedCall: DecodedCall;
}
