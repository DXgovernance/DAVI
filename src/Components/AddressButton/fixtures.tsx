import { AddressButtonProps } from './types';

export const fullProps: AddressButtonProps = {
  address: '0x79706C8e413CDaeE9E63f282507287b9Ea9C0928',
  transactionsCounter: 0,
  onClick: jest.fn(),
};

export const partialProps: AddressButtonProps = {
  address: '0x79706C8e413CDaeE9E63f282507287b9Ea9C0928',
};
