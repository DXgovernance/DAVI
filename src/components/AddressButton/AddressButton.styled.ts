import styled from 'styled-components';
import { IconButton } from 'components/primitives/Button';
import { Badge } from 'components/Badge';

export const IconHolder = styled.span`
  display: flex;
  justify-content: center;

  @media only screen and (min-width: 768px) {
    margin-right: 0.3rem;
  }

  img {
    border-radius: 50%;
    margin-right: 0;
  }
`;

export const StyledAddressButton = styled(IconButton)`
  margin-top: 0;
  margin-bottom: 0;
  padding: 0.3rem;

  @media only screen and (min-width: 768px) {
    padding: 0.3rem 0.5rem;
  }

  /* hover state for when having child Badge */
  &:hover,
  &:active {
    ${Badge} {
      background-color: ${({ theme }) => theme.colors.background};
      color: ${({ theme }) => theme.colors.primary};
    }
  }
`;

export const AddressText = styled.span`
  margin-left: 0.2rem;
  margin-right: 0.3rem;
`;
