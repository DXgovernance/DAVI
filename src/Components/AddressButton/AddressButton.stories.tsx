import { ComponentStory, ComponentMeta } from '@storybook/react';
import AddressButton from 'Components/AddressButton/AddressButton';
import '@testing-library/jest-dom';
import { fullProps, partialProps } from './fixtures';

export default {
  title: 'Components/AddressButton',
  component: AddressButton,
  argTypes: {
    address: {
      description: 'Proposal creator address',
    },
    transactionsCounter: {
      description: 'Number of transactions of the proposal',
    },
  },
} as ComponentMeta<typeof AddressButton>;

const Template: ComponentStory<typeof AddressButton> = args => (
  <AddressButton {...args} />
);

export const WithTransactions = Template.bind({});
WithTransactions.args = fullProps;

export const WithoutTransactions = Template.bind({});
WithoutTransactions.args = partialProps;

export const Loading = Template.bind({});
Loading.args = {};

