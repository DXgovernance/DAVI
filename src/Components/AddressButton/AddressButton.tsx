import { isDesktop } from 'react-device-detect';
import styled from 'styled-components';
import React from 'react';

import { IconButton } from 'old-components/Guilds/common/Button';
import useENSAvatar from 'hooks/Guilds/ether-swr/ens/useENSAvatar';
import Avatar from 'old-components/Guilds/Avatar';
import { shortenAddress } from 'utils';
import { Badge } from 'old-components/Guilds/common/Badge';
import { Loading } from 'Components/Primitives/Loading';

import { AddressButtonProps } from 'Components/AddressButton/types';

const IconHolder = styled.span`
  display: flex;
  justify-content: center;

  @media only screen and (min-width: 768px) {
    margin-right: 0.3rem;
  }

  img {
    border-radius: 50%;
    margin-right: 0;
  }
`;

const StyledAddressButton = styled(IconButton)`
  margin-top: 0;
  margin-bottom: 0;
  padding: 0.3rem;

  @media only screen and (min-width: 768px) {
    padding: 0.3rem 0.5rem;
  }

  /* hover state for when having child Badge */
  &:hover,
  &:active {
    ${Badge} {
      background-color: ${({ theme }) => theme.colors.background};
      color: ${({ theme }) => theme.colors.primary};
    }
  }
`;

const AddressText = styled.span`
  margin-left: 0.2rem;
  margin-right: 0.3rem;
`;

const AddressButton: React.FC<AddressButtonProps> = ({
  address,
  transactionsCounter,
  onClick,
}) => {
  const { ensName, imageUrl } = useENSAvatar(address, 1);

  return (
    <StyledAddressButton variant="secondary" onClick={onClick} iconLeft>
      <IconHolder>
        {address ? (
          <Avatar src={imageUrl} defaultSeed={address} size={24} />
        ) : (
          <Loading
            loading
            text
            skeletonProps={{ circle: true, width: '24px', height: '24px' }}
          />
        )}
      </IconHolder>
      {isDesktop && (
        <AddressText>
          {ensName || address ? (
            shortenAddress(address)
          ) : (
            <Loading loading text />
          )}
        </AddressText>
      )}
      {transactionsCounter ? (
        <Badge size="25">{transactionsCounter}</Badge>
      ) : null}
    </StyledAddressButton>
  );
};

export default AddressButton;
