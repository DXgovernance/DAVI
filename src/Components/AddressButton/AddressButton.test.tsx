import '@testing-library/jest-dom';
import AddressButton from './AddressButton';
import { AddressButtonProps } from 'Components/AddressButton/types';
import { render } from '../../utils/tests';

jest.mock('hooks/Guilds/ether-swr/ens/useENSAvatar', () => ({
  __esModule: true,
  default: () => ({
    avatarUri: 'test',
    imageUrl: 'test',
    ensName: 'test.eth',
  }),
}));

const fullProps: AddressButtonProps = {
  address: '0x79706C8e413CDaeE9E63f282507287b9Ea9C0928',
  transactionsCounter: 0,
  onClick: jest.fn(),
};

const partialProps: AddressButtonProps = {
  address: '0x79706C8e413CDaeE9E63f282507287b9Ea9C0928',
};

test('AddressButton renders properly with full props', async () => {
  const { container } = render(<AddressButton {...fullProps} />);
  expect(container).toMatchSnapshot();
});

test('AddressButton renders properly with partial props', async () => {
  const { container } = render(<AddressButton {...partialProps} />);
  expect(container).toMatchSnapshot();
});
