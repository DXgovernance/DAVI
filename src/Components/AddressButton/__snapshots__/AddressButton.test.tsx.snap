// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressButton Should render properly with full props 1`] = `
<div>
   
  <button
    class="sc-dlfnuX sc-hKgJUU sc-fubCzh fUlBHn eRtASc jmvQcZ"
    type="submit"
  >
    <span
      class="sc-iBPTik lkybgw"
    >
      <img
        alt="Avatar"
        class="sc-eCstlR sc-jSgvzq ebJAYl bmPvYM"
        data-testid="avatar"
        size="24"
        src="test"
      />
    </span>
    <span
      class="sc-pGacB hgMCTu"
    >
      0x7970...0928
    </span>
    <div
      class="sc-gKseQn eOEXWN"
      size="25"
    >
      5
    </div>
  </button>
   
</div>
`;

exports[`AddressButton Should render properly with partial props 1`] = `
<div>
   
  <button
    class="sc-dlfnuX sc-hKgJUU sc-fubCzh fUlBHn eRtASc jmvQcZ"
    type="submit"
  >
    <span
      class="sc-iBPTik lkybgw"
    >
      <img
        alt="Avatar"
        class="sc-eCstlR sc-jSgvzq ebJAYl bmPvYM"
        data-testid="avatar"
        size="24"
        src="test"
      />
    </span>
    <span
      class="sc-pGacB hgMCTu"
    >
      0x7970...0928
    </span>
  </button>
   
</div>
`;
